front/app/api/analysis_insights/useAnalysisInsightsById.ts

```diff
@@ -0,0 +1,24 @@
+import { UseQueryOptions, useQueries } from '@tanstack/react-query';
+import { IInsights } from './types';
+import insightsKeys from './keys';
+import { fetchInsights } from './useAnalysisInsights';
+
+type AnalysisInsightsWithIdsReturnType = UseQueryOptions<IInsights>[];
+
+const useAnalysisInsightsWithIds = (analysisIds?: string[]) => {
+  const queries = analysisIds
+    ? analysisIds.map((analysisIds) => ({
+        queryKey: insightsKeys.list({
+          analysisId: analysisIds,
+          bookmarked: true,
```

Comment:
It's now not obvious from the name of the hook why this one only shows bookmarked insights. Maybe `bookmarked` can be a parameter?

Reply:
Yes, makes sense for it to be a parameter 👍